### DISCUSSION TITLE
Weird Bouncing issue with Drawer.

### DISCUSSION THREAD

**Discussion by premdasvm** [react-navigation/react-navigation#10485] [Q&A]
Having a weird issue with my drawer, any idea on how to fix it?

https://user-images.githubusercontent.com/14838550/161826217-74e36ff5-5360-4237-ba57-470a75a2497d.mp4

**Comment by markodvornik** [comment]
There are 2 scenarios depending on how the drawer was opened:

1. by _Button onPress open event_: drawer bounces back to open state - repeated press closes the drawer
2. by _swipe to open event_: drawer stays open and changes the underlying screen - repeated press closes the drawer in & renders the screen that was pressed on that second press (it could be any screen, not just the one pressed previously)

Also , it looks like if screen changing is done slowly bouncing doesn't occur. Fast changes eventualy auto fire open event.

Logged timings of state changes:

```
drawerStatus open    0.160 <-- Open drawer
drawerStatus closed +1.059 <-- Pressed
drawerStatus open   +1.199 <-- Auto fired
```

**Reply by premdasvm** [reply]
hey, Actually this workaround fixed my issue , https://github.com/react-navigation/react-navigation/issues/10495#issue-1198722521
